pkg/vendir/fetch/git/sync.go

```diff
@@ -46,9 +51,24 @@ func (d Sync) Sync(dstPath string, tempArea ctlfetch.TempArea) (ctlconf.LockDire
 
 	defer os.RemoveAll(incomingTmpPath)
 
+	sum := sha256.New()
+	if _, err := sum.Write([]byte(d.opts.URL)); err != nil {
+		if err != nil {
+			return gitLockConf, err
+		}
+	}
+
+	cacheID := string(sum.Sum(nil))
```

Comment:
The nested `if err != nil` is redundant.

What about simplifying the code like this:

```suggestion
	cacheID := fmt.Sprintf("%x", sha256.Sum256([]byte(d.opts.URL)))
```